### ISSUE TITLE
Publish a Blazor app using standalone method in a virtual path (problems)

### ISSUE THREAD

**Issue by marcelovidigal** [aspnet/AspNetCore.Docs#13715]
I followed the official documentation (https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/blazor/client-side?view=aspnetcore-3.0), but the steps for publishing a blazor client side app are unclear.

I would like to publish to IIS.

The modification in the base tag (`<base href="/">`) has a bug.

The suggested modifications to webconfig (parent) and webconfig (app folder) do not work, and in my case the tags to change in parent web.config didn't even exist.

First try, I created a virtual folder, turned the folder into an application, wanted to access `http://site/app`, ran the `dotnet publish -c release` command, and copied the dist folder (`\bin\release\netstandard2.0\publish\app\dist`) from the release to the application's virtual folder. Unsuccessfully. Routes do not work and the application always routes urls with reference to the root and not the virtual folder.

Second try, I modified the base <base href> tag by pointing to the virtual folder (`<base href="/App/"`). First problem, a lot of errors, after a lot of debugging I realized that the tag like this (`<base href="/app/">`), without the capitalization, worked, I imagine it's a bug. The paths to the pages continued without referencing the application folder .

Third try, I copied the whole folder of the publication (`\bin\release\netstandard2.0\publish\`), as the second try, also ran the application, with the same capitalization problem (corrected before), and same problem of paths not including the virtual folder (not corrected). 

What is the correct approach? Copy entire publish folder? Or just the dist folder? Or just static content and webconfig? The documentation doesn't make that clear either, just copying the dist folder causes a lot of headaches with the application paths. The most straightforward approach was to copy all the contents of the publish folder, and the most correct, static content (assembly folder) and webconfig with the rewrites of the routes.

Still, the application loads, but the page paths do not include the virtual folder.

Publishing directly to the root works perfectly, but using a virtual folder is quite problematic, and even worse in IIS, the solution for now has been to directly reference the virtual folder in the application page paths if you want to use a virtual directory.

More detailed documentation of ways of publishing client-side apps would be of great help.

**Comment by guardrex** [comment]
Hello @marcelovidigal ... I only just saw your issue today when I was looking over engineering issues. 

Side note for the future in case you run into any new docs issues: There's a **This Page** feedback button at the bottom of each topic that opens an issue for the docs team on this repo. It also adds document metadata to the issue to help us track the work. I just mention this in case you need to open future issues for the docs team.

Sorry that you ran into trouble with the doc. The Blazor docs are a *work in progress* during preview. We're taking feedback and making improvements for the final release. I'm sorry you had trouble, but we'll get it fixed up. *Future generations will thank us!* :smile:

This is going to get a little 📘 *book-ish* 📘 here, but that's because there are several deployment scenarios, and a couple of them overlap in IIS.

Some points on the content after looking at your remarks and the topic:

* The word "virtual" is a bit overloaded: The topic shouldn't guide the use of an IIS virtual directory. Only sub-apps are supported with ASP.NET Core.

  Further down, the text calls out "IIS sub-app," but it's a fine point in the midst of a lot of content. Guidance not to use an IIS vdir is in our IIS topic, but I didn't think to mirror that information here. Perhaps, I was thinking at the time that if a dev uses the **Hosted** Blazor client-side template that they'll get into our ASP.NET Core *Host and Deploy* > *IIS Overview*, which is cross-linked, and discover the prohibition on IIS vdirs that way.

  "Virtual" here merely refers to the virtual URL (path). Where the topic mentions "virtual folder," I see how it's confusing/misleading.

  I also have a concern about the guidance in the *App base path* section: Where the content covers  modifying the *web.config* for a Blazor client-side IIS sub-app, I think that might work better split out as a different deployment type/strategy/scenario. It looks like from your comments that it misled you where it's currently described and explained.

  The reason I'm thinking that there are *TWO* scenarios here is that someone might use the Blazor client-side **Hosted** template to get a shiny new template-based ASP.NET Core app with a Blazor client-side app to serve ... **OR** ... they might already have a fully working ASP.NET Core app that they would like to embed a Blazor client-side app into as an IIS sub-app. See what I mean? Figuring out how to cover these scenarios will take a bit more consideration. 🤔

* I also see from your remarks that the distinction and guidance for the **Hosted** deployment (Blazor client-side served by an ASP.NET Core app) versus the **Standalone** deployment (Blazor client-side deployed as a set of static files) wasn't clear enough. These deployment scenarios are covered in two sections called out by the *Deployment* section, but the *Deployment* section should probably be moved down in the topic and be a parent section of the deployment sections. I'll work on that, but to quickly describe them here ...
  * The **Hosted** scenario has an ASP.NET Core app hosting the Blazor client-side app: You publish and deploy that just like any other ASP.NET Core app. The regular IIS guidance for ASP.NET Core apps applies in that scenario, and that guidance is cross-linked.
  * The client-side app deployed as a set of static files, which we call the **Standalone** deployment, doesn't rely upon an ASP.NET Core app to serve the Blazor app. If such an app is to be hosted by IIS, IIS is acting as a static file server. We cross-link the IIS static file server doc in the IIS doc set for that.

In summary for now until I get to updating the doc, decide if you want an ASP.NET Core app to host your Blazor-client side app.
* If *Yes* ... Use the **Hosted** Blazor client-side template. Publish and deploy just like any other ASP.NET Core app.
* If *No* ... then do you want IIS to host the Blazor client-side app's static files?
  * *Yes* ... Follow our *Standalone deployment* > *IIS* instructions in the topic. You'll use the *publish* folder contents and the *web.config*.
  * *No* ... Take the published contents of the *dist* folder. That's your static Blazor site for deployment to some other static file server.

I should have some time this weekend to address these issues. I'll ping you on the PR, and we can discuss the updates from there.
